lib/Service/FavoritesService.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2024 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Tables\Service;
+
+use OCA\Tables\AppInfo\Application;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCP\Cache\CappedMemoryCache;
+use OCP\DB\Exception;
+use OCP\IDBConnection;
+
+class FavoritesService {
+
+	private IDBConnection $connection;
+	private PermissionsService $permissionsService;
+	private ?string $userId;
+	private CappedMemoryCache $cache;
+
+	public function __construct(
+		IDBConnection $connection,
+		PermissionsService $permissionsService,
+		?string $userId
+	) {
+		$this->connection = $connection;
+		$this->permissionsService = $permissionsService;
+		$this->userId = $userId;
+		$this->cache = new CappedMemoryCache();
+	}
+
+	public function isFavorite(int $nodeType, int $id): bool {
+		$cacheKey = $this->userId . '_' . $nodeType . '_' . $id;
```

Comment:
userId does not need to be in the cacheKey. This confused me a bit, for it appears the cache could be in use for different users (and then clearing everything in the next line would be too much). Naming things ;)

Can change, not must change.

Reply:
Adjusted but with a comment on the constructor to avoid someone changing this to a shared cache without thinking about side effects.